doc/bakugo_spec.md

```diff
@@ -546,15 +544,15 @@ The _scope_ of a declared identifier is the extent of source text in which the i
 constant, type, variable, function, label, or package.
 
 
-Go is lexically scoped using _blocks_:
+Go is lexically scoped:
 
 1. The scope of a _predeclared identifier_ is the universe block.
-2. The scope of an identifier denoting a function parameter, or result variable is the function body.
-3. The scope of a constant or variable identifier declared inside a function begins at the end of the ConstSpec or VarSpec and ends at the end of the innermost containing block.
-4. The scope of a type identifier declared inside a function begins at the identifier in the TypeSpec and ends at the end of the innermost containing block.
+1. The scope of an identifier denoting a function result variable is the function body.
```

Comment:
Does this mean that an identifier denoting a function result variable cannot be re-declared in the function body?

Reply:
Yes, that is correct. A function result variable is declared in the function definition and whatever assigned to it is returned. So it doesn't make sense to re-declare it anywhere in the body because the binding created at the definition is used at the end of the body.

Reply:
I never liked the idea of the return variable being declared in the definition. I don't mind removing that feature entirely xD

Reply:
> So it doesn't make sense to re-declare it anywhere in the body because the binding created at the definition is used at the end of the body.

Yes, I was wondering that. Cool! Implementing it would be fun.

Reply:
> I never liked the idea of the return variable being declared in the definition. I don't mind removing that feature entirely xD

Yes, I had something similar on my mind. xD

Reply:
Let's remove it then?

Reply:
Okay, Let's remove them.

Reply:
It's done in 4bce513

Reply:
Nice